replicator/src/main/scala/com/evolutiongaming/kafka/journal/replicator/Batch.scala

```diff
@@ -98,6 +109,15 @@ private[journal] object Batch {
                     val delete = deleteOf(a.to, b.origin orElse a.origin, b.version orElse a.version)
                     delete :: tail
                   }
+
+                  val cleanTail: List[Batch] = tail.map {
+                    case appends: Appends => dropDeleted(appends, a)
+                    case delete: Delete   => delete.some
+                    case purge: Purge     => purge.some
+                  }.flattenOption
+
+                  val delete = deleteOf(a.to, b.origin orElse a.origin, b.version orElse a.version)
```

Comment:
I find it a bit awkward that we do the following:
```scala
if (a.to > b.to) {
  ..., b.origin orElse a.origin, b.version orElse a.version...
} else {
  ..., b.origin orElse a.origin, b.version orElse a.version...
}
```
If I understood it correctly, it means that sometimes the older origin/version will be used, and sometimes a newer one.

It does not matter, probably, and was like that even before this PR, but I just wanted to comment on it.

Reply:
the logic os very convoluted here - idea is that `delete` could be the first action in stream, which "inits" journal - and we want (have?) to preserve the `origin` and `version` of journal's first event - there are several test-cases which test this for `origin` (but not for `version`).

I am not sure, why we need that, but maybe idea was that it would allow us to (just guessing):
* `origin` - find the pod, which created the journal
* `version` - provide an ability to make an educated guess on how many old journals we have - like if we can drop backwards compatibility for `kafka-journal` specific data